Libraries/VaahStripe.php

```diff
@@ -0,0 +1,446 @@
+            ]);
+
+            $response['status'] = 'success';
+            $response['data']   = $product;
+
+        }catch(\Exception $e)
+        {
+            $response['status']     = 'failed';
+            $response['errors'][]   = $e->getMessage();
+        }
+
+        return $response;
+
+    }
+    //----------------------------------------------------------
+
+    public function createPrice(Request $request)
+    {
+        $inputs = $request->all();
+
+        $rules = array(
+            'product_id'    => 'required',
+            'amount'        => 'required',
+            'currency'      => 'required',
+            'interval'      => 'required'
+        );
+
+        $validator = \Validator::make( $inputs, $rules);
+        if ( $validator->fails() ) {
+
+            $errors             = errorsToArray($validator->errors());
+            $response['status'] = 'failed';
+            $response['errors'] = $errors;
+            return $response;
+        }
+
+        try{
+
+            $product = Stripe::prices()->create([
+                'unit_amount'   => $inputs['amount'],
+                'currency'      => $inputs['currency'],
+                'product'       => $inputs['product_id'],
+                'recurring'     => [
+                    'interval'  => $inputs['interval']
+                ]
+            ]);
+
+            $response['status'] = 'success';
+            $response['data']   = $product;
+
+        }catch(\Exception $e)
+        {
+            $response['status']     = 'failed';
+            $response['errors'][]   = $e->getMessage();
+        }
+
+        return $response;
+
+    }
+    //----------------------------------------------------------
+
+    public function findProductByName($name)
+    {
+
+        if(!$name){
+            $response['status'] = 'failed';
+            $response['errors'] = 'The name field is required.';
+            return $response;
+        }
+
+        try{
+
+            $product_val  = null;
+
+            $data = [
+                'active' => true
+            ];
+
+            $products = Stripe::products()->all($data);
+
+            foreach ($products['data'] as $product){
+
+                if($product['name'] == $name){
+                    $product_val =  $product;
+                }
+
+            }
+
+            if(!$product_val){
+                $response['status']     = 'failed';
+                $response['errors'][]   = 'No Product Found';
+                return $response;
+            }
+
+            $response['status'] = 'success';
+            $response['data']   = $product_val;
+
+
+        }catch(\Exception $e)
+        {
+            $response['status']     = 'failed';
+            $response['errors'][]   = $e->getMessage();
+        }
+
+        return $response;
+
+    }
+    //----------------------------------------------------------
+
+    public function findPriceByProductId($product_id, $by = null, $value = null)
```

Comment:
@wri-vikram-wri231 `$value` should come first, and then `by` and the default value of `by` should be `amount`